robot_calibration/include/robot_calibration/capture_features.h

```diff
@@ -0,0 +1,124 @@
+#ifndef ROBOT_CALIBRATION_CAPTURE_FEATURES_H
+#define ROBOT_CALIBRATION_CAPTURE_FEATURES_H
+
+
+/**
+* @brief      Verifies the required parameters have been set and, if not, set default values.
+*
+* @param[in]  node  node handle reference
+*
+* @return     bool indicating whether the parameters are all set
+*/
+bool check_parameters(ros::NodeHandle& node);
+
+
+/**
+* @brief      Creates a string for the absoulte path for a file in the home directory.
+*
+* @param[in]  local_dir     name of the local directory.
+*
+* @return     string representing the absolute path to the file
+*/
+std::string get_absolute_directory(std::string local_dir);
+
+
+/**
+* @brief      Creates a map of the available feature finders
+*
+* @param[in]  nh        node handle reference.
+* @param[in]  finders   the location to store the map of feature finders
```

Comment:
This is an output parameter, consider putting it in the return value.  If you have C++17 in this library you can could return something like this:

```c++
std::optional<robot_calibration::FeatureFinderMap> get_feature_finders(const ros::NodeHandle& nh);
```